File: app/src/main/java/com/marked/pixsee/data/database/PixyDatabase.java
package com.marked.pixsee.data.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

/**
 * Created by Tudor Pop on 26-Jan-16.
 */
public class PixyDatabase extends SQLiteOpenHelper {

	private static PixyDatabase instance;


	private PixyDatabase(Context context) {
		super(context, DatabaseContract.DATABASE_NAME, null, DatabaseContract.DATABASE_VERSION);
	}

	public static PixyDatabase getInstance(Context context) {
		if (instance == null)
			instance = new PixyDatabase(context);
		return instance;
	}


	@Override
	public void onCreate(SQLiteDatabase db) {
		for (String it : DatabaseContract.CREATE_TABLE_ARRAY)
			db.execSQL(it);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
		for (String it : DatabaseContract.DELETE_TABLE_ARRAY)
			db.execSQL(it);
		onCreate(db);
	}
}

File: app/src/main/java/com/marked/vifo/model/database/PixyDatabase.kt
package com.marked.vifo.model.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import org.jetbrains.anko.db.ManagedSQLiteOpenHelper

/**
 * Created by Tudor Pop on 26-Jan-16.
 */
class PixyDatabase private constructor(context: Context) : ManagedSQLiteOpenHelper(context, DatabaseContract.DATABASE_NAME, version = DatabaseContract.DATABASE_VERSION) {
	companion object {
		private var instance: PixyDatabase? = null

		@Synchronized
		fun getInstance(context: Context): PixyDatabase {
			if (instance == null)
				instance = PixyDatabase(context.applicationContext)
			return instance!!
		}
	}

	override fun onCreate(db: SQLiteDatabase) {
		// Here you create tables (more info about that is below)
		for (it in DatabaseContract.CREATE_TABLE_ARRAY)
			db.execSQL(it)
	}

	override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
		// Here you can upgrade tables, as usual
		for (it in DatabaseContract.DELETE_TABLE_ARRAY)
			db.execSQL(it);
		onCreate(db);
	}

}

val Context.database: PixyDatabase
	get() = PixyDatabase.getInstance(applicationContext)
